File: lab2/gen.py
import random
import time


def generate():
    try:
        random.seed(time.time())

        bits = ""

        for _ in range(128):
            bits += str(random.randint(0, 1))

        with open("py_sequence.txt", "w", encoding="utf-8") as f:
            f.write(bits)

        print("Python: файл успешно создан")

    except Exception as e:
        print(f"Ошибка Python генерации: {e}")


if __name__ == "__main__":
    generate()

File: lab2/gen.cpp
#include <iostream>
#include <fstream>
#include <random>
#include <ctime>

using namespace std;

void generate()
{
    try
    {
        ofstream file("cpp_sequence.txt");

        if (!file)
        {
            throw runtime_error("Не удалось открыть файл");
        }

        mt19937 gen(time(nullptr));
        uniform_int_distribution<int> dist(0, 1);

        for (int i = 0; i < 128; i++)
        {
            file << dist(gen);
        }

        file.close();

        cout << "C++: файл успешно создан" << endl;
    }
    catch (const exception &e)
    {
        cout << "Ошибка C++ генерации: " << e.what() << endl;
    }
}

int main()
{
    generate();
    return 0;
}
